File: src/main/java/iot/tetracubered/config/MessagingConfig.java
package iot.tetracubered.config;

import org.eclipse.microprofile.config.inject.ConfigProperty;

public interface MessagingConfig {

    @ConfigProperty(name = "host")
    String host();

    @ConfigProperty(name = "port")
    Integer port();

    @ConfigProperty(name = "user")
    String user();

    @ConfigProperty(name = "password")
    String password();

    @ConfigProperty(name = "topics")
    TopicsConfig topics();
}

File: src/main/kotlin/iot/tetracubered/config/MessagingConfig.kt
package iot.tetracubered.config

import org.eclipse.microprofile.config.inject.ConfigProperty

interface MessagingConfig {

    @ConfigProperty(name = "host")
    fun host(): String

    @ConfigProperty(name = "port")
    fun port(): String

    @ConfigProperty(name = "user")
    fun user(): String

    @ConfigProperty(name = "password")
    fun password(): String

    @ConfigProperty(name = "topics")
    fun topics(): TopicsConfig
}